libraries/stdlib/jvm/src/kotlin/io/files/FileTreeWalk.kt

```diff
@@ -77,7 +92,13 @@ public class FileTreeWalk private constructor(
         }
 
         override fun computeNext() {
-            val nextFile = gotoNext()
+            val nextFile: File?
+            if (algorithm.equals(WalkAlgorithm.BFS)) {
```

Comment:
It's probably nicer to use `if` as an expression here:

            val nextFile = if (algorithm.equals(WalkAlgorithm.BFS)) {
                ++currentDepth
                gotoNextBfs()
            } else {
                gotoNextDfs()
            }

Reply:
Agree.

Reply:
> It's probably nicer to use `if` as an expression here:
> 
> ```
>         val nextFile = if (algorithm.equals(WalkAlgorithm.BFS)) {
>             ++currentDepth
>             gotoNextBfs()
>         } else {
>             gotoNextDfs()
>         }
> ```

Done.